### PR TITLE
Add documentation for dynamic router generate event

### PR DESCRIPTION
**Pull request by benglass**
Documentation for https://github.com/symfony-cmf/Routing/pull/131

### CODE REVIEW

File: components/routing/dynamic.rst [hunk 1]
```diff
@@ -44,6 +44,8 @@ process, depending on which method is used:
 * **cmf_routing.pre_dynamic_match_request** (Dispatched at the beginning of the
   ``matchRequest`` method. In the context of the Symfony2 full stack framework,
   only this event will be triggered.)
+* **cmf_routing.pre_dynamic_generate** (Dispatched at the beginning of the
+  ``generate`` method)
```
Comment (dbu) [inline components/routing/dynamic.rst]: can you please add a `..since` directive to say that the generate event is added in routing 1.4?

Comment (WouterJ) [inline components/routing/dynamic.rst]: you mean `.. versionadded:: ` :)

**Comment by dbu** [comment]
@benglass can you please update this pull request so that we can merge the doc? the missing versionadded thing, otherwise its all good.

File: components/routing/dynamic.rst (hunk 1 above)
Comment (dbu) [inline components/routing/dynamic.rst]: we should also mention here that you can change the route to be generated by calling setRoute on this event. (see https://github.com/symfony-cmf/Routing/pull/135)

**Comment by dbu** [comment]
expanded doc in https://github.com/symfony-cmf/symfony-cmf-docs/commit/a80cbe9b6da71de7828242e66a888645e7e7a75a

**Pull request closed without merging**